src/index.rs

```diff
@@ -641,4 +670,46 @@ mod test {
 			}
 		}
 	}
+
+	#[cfg(feature = "bench")]
+	fn bench_find_entry_internal<
+		F: Fn(&IndexTable, u64, usize, &[u8; CHUNK_LEN]) -> (Entry, usize),
+	>(
+		b: &mut Bencher,
+		f: F,
+	) {
+		let table =
+			IndexTable { id: TableId(18), map: RwLock::new(None), path: Default::default() };
+		let mut chunk = [0u8; 512];
+		let mut keys = [0u64; 64];
+		let mut rng = rand::thread_rng();
+		for i in 0..64 {
+			keys[i] = rng.gen();
+			let partial_key = Entry::extract_key(keys[i], 18);
+			let e = Entry::new(Address::new(0, 0), partial_key, 18);
+			IndexTable::write_entry(&e, i, &mut chunk);
+		}
+
+		let mut index = 0;
+
+		b.iter(|| {
+			let i = index % 64;
+			let x = f(&table, keys[i], 0, &chunk).1;
+			assert_eq!(x, i);
+			index += 1;
```

Comment:
Would make sense to me to have the 64 query under iter (so nb iter don't interfer (actually with this use case it probably does not really matter)).

Reply:
Under real workload chunks are never full. I'd guess average occupancy is about 33%. With just a few entries base variant will probably be faster, while with 64 entries simd will probably be faster.  This cycle averages it.